src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -277,6 +277,94 @@ module Base = struct
     in
     Boolean.Assert.any [Boolean.not is_user_command; verifies]
 
+  let check_timing ~account ~txn_amount ~txn_global_slot =
+    (* calculations should track Transaction_logic.validate_timing *)
+    let open Account.Poly in
+    let open Account.Timing.As_record in
+    let { is_timed
+        ; initial_minimum_balance
+        ; cliff_time
+        ; vesting_period
+        ; vesting_increment } =
+      account.timing
+    in
+    let%bind before_cliff =
+      Global_slot.Checked.(txn_global_slot < cliff_time)
+    in
+    let int_of_field field =
+      Snarky_integer.Integer.constant ~m
+        (Bigint.of_field field |> Bigint.to_bignum_bigint)
+    in
+    let zero_int = int_of_field Field.zero in
+    let balance_to_int balance =
+      Snarky_integer.Integer.of_bits ~m @@ Balance.var_to_bits balance
+    in
+    let sub_min_zero int1 int2 =
+      (* actual difference, or zero *)
+      let open Snarky_integer.Integer in
+      let subtrahend = min ~m int1 int2 in
```

Comment:
I think we can do something more efficient (and simpler to read) here, emitting a flag when the bit expansion isn't equal. I'll put something together tomorrow.